floodlight/io/kinexon.py

```diff
@@ -0,0 +1,349 @@
+                        if (
+                            line[parameter_links[identifier]]
+                            not in team_infos["0"][identifier]
+                        ):
+                            team_infos["0"][identifier].append(
+                                line[parameter_links[identifier]]
+                            )
+
+    # sort dict
+    team_infos = dict(sorted(team_infos.items()))
+
+    # estimate framerate
+    timestamps = list(set(t))
+    timestamps.sort()
+    timestamps = np.array(timestamps)
+    minimum_time_step = np.min(np.diff(timestamps))
+    frame_rate = int(1000 / minimum_time_step)
+
+    number_of_frames = int((timestamps[-1] - timestamps[0]) / (1000 / frame_rate))
+    t_null = timestamps[0]
+
+    return team_infos, number_of_frames, frame_rate, t_null
+
+
+def create_links_from_meta_data(
+    team_infos: Dict[str, Dict[str, List[str]]], identifier: str = None
+) -> Dict[str, Dict[str, int]]:
+    """Creates a dictionary from the team_info dict linking the identifier to the xID.
+
+    Parameters
+    ----------
+    team_infos: Dict[str, Dict[str, List[str]]],
+        Nested dictionary that stores information about the ids from every identifier
+        in every group. Identifier are sensor_id, mapped_id, name. If the respective
+        id-type is not in the recorded parameters, it is not listed in team_infos.
+        'team_info[group][identifier] = [id1, id2, ..., idn]'
+    identifier: str
+        Column-name of personal identifier in Kinexon.csv-file.
+        Can be one of:
+            - 'sensor_id'
+            - 'mapped_id'
+            - 'name'
+
+    Returns
+    -------
+    links: Dict[str, Dict[str, int]]
+        Link-dictionary of the form `links[group][identifier-ID] = xID`.
+    """
+
+    for player_id in ["name", "mapped_id", "sensor_id"]:
+        if identifier is None:
+            if player_id in list(team_infos.values())[0]:
+                identifier = player_id
+                break
+
+    links = {}
+    for gID in team_infos:
+        links.update(
+            {gID: {ID: xID for xID, ID in enumerate(team_infos[gID][identifier])}}
+        )
+
+    return links
+
+
+def read_kinexon_file(filepath_data: Union[str, Path], identifier: str = None) -> XY:
+    """Parse Kinexon files and extract position data.
+
+    Kinexon's local positioning system delivers one .csv file containing the position
+    data. This function provides a high-level access to Kinexon data by parsing "the
+    full file" given the path to the file.
+
+    Parameters
+    ----------
+    filepath_data: str of pathlib.Path
+        Full path to Kinexon.csv-file.
+    identifier: str
+        Column-name of personal identifier in Kinexon.csv-file.
+        Can be one of:
+            - 'sensor_id'
+            - 'mapped_id'
+            - 'name'
+
+    Returns
+    -------
+    positions: XY
+        XY-Object for the whole game. The order of groups is ascending according to
```

Comment:
This is not clear to me, do you intend to pass a list of XY, or do you concatenate the data?